modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +548,48 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['BehaviouralSubproject'])
+            && $_REQUEST['BehaviouralSubproject'] != ''
```

Comment:
Typically instead of `isset($var) && $var != ''` we use `!empty($var)`. This has the same truth value and has the benefit of being clearer and shorter.

Since this is one code convention we do actually have, please enforce it here 😅